src/index.ts

```diff
@@ -132,15 +134,59 @@ function addSegmentFromDirectionsResponse(previousLngLat: LngLat, e: MapMouseEve
     });
 }
 
-function addSegmentFromStraightLine(previousLngLat: LngLat, e: MapMouseEvent): void {
-  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, e.lngLat);
+function addSegmentFromStraightLine(previousLngLat: LngLat, lngLat: LngLat, animate = true): void {
+  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, lngLat);
 
-  animationService.animateSegment(newSegment);
-  const marker = addMarker(e.lngLat, false);
+  if (animate) animationService.animateSegment(newSegment);
+  const marker = addMarker(lngLat, false);
   currentRun.addSegment(newSegment, marker);
   updateLengthElement();
 }
 
+function runToJson(run: CurrentRun): string {
+  if (run === undefined) return "{}";
+  let runJSON: {[name:string]: any} = {
+    start: {
+      lng: run.start.lngLat.lng,
+      lat: run.start.lngLat.lat,
+    },
+    distance: run.distance,
+    segments: [],
+    followRoads
+  }
+  for (let i in run.segments) {
+    runJSON.segments.push({
+      lng: run.segments[i].lngLat.lng,
+      lat: run.segments[i].lngLat.lat,
+      followsRoads: run.segments[i].followsRoads
+    });
+  }
+  return JSON.stringify(runJSON);
+}
+
+function jsonToRun(json: string) {
+  let runJSON = JSON.parse(json);
```

Comment:
because `JSON.parse(...` can go south and throw errors, may be good to wrap this in a try/catch that at least prints the error to the console. We can later add in a nice little toast or something to indicate to the user something has gone wrong

Reply:
also, having the interface above would let us be able to give a type to `runJSON`, and we could be confident about the fields we're accessing